src/crypto/equihash.cpp

```diff
@@ -247,7 +246,7 @@ std::set<std::vector<eh_index>> Equihash::BasicSolve(const eh_HashState& base_st
 
 bool Equihash::IsValidSolution(const eh_HashState& base_state, std::vector<eh_index> soln)
 {
-    eh_index soln_size { pow(2, k) };
+    eh_index soln_size { 1u << k };
```

Comment:
Yes, there's a compiler warning if it's not explicitly unsigned (probably for the case where k=31)